app/routes.py

```diff
@@ -0,0 +1,345 @@
+    customer_rentals_response = []
+    for customer_rental in customer_rentals:
+        video = Video.query.get(customer_rental.video_id)
+        customer_rentals_response.append({
+            "release_date" : video.release_date,
+            "title" : video.title,
+            "due_date" : customer_rental.due_date
+        })
+    return jsonify(customer_rentals_response)
+
+
+@videos_bp.route("/<video_id>/rentals", methods=["GET"])
+def list_video_rentals(video_id):
+    video = Video.query.get(video_id)
+    if video is None:
+        return jsonify({
+            "message" : f"Video {video_id} was not found"
+        }), 404
+    video_rentals = Rental.query.filter_by(video_id=video_id, checked_in_status=False).all()
+    video_rentals_response = []
+    for video_rental in video_rentals:
+        customer = Customer.query.get(video_rental.customer_id)
+        video_rentals_response.append({
+            "due_date" : video_rental.due_date,
+            "name" : customer.name,
+            "phone" : customer.phone,
+            "postal_code" : customer.postal_code
+        })
+    return jsonify(video_rentals_response)
+
+
+@videos_bp.route("", methods=["GET", "POST"])
+def handle_videos():
+    if request.method == "GET":
+        videos = Video.query.all()
+        videos_response = []
+        for video in videos:
+            videos_response.append({
+                "id": video.id,
+                "title": video.title,
+                "release_date": video.release_date,
+                "total_inventory": video.total_inventory
+            })
+        
+        return jsonify(videos_response), 200
+
+    elif request.method == "POST":
+        request_body = request.get_json()
+        title = request_body.get("title")
+        release_date = request_body.get("release_date")
+        total_inventory = request_body.get("total_inventory")
+
+        if not title:
+            return jsonify({
+                "details" : "Request body must include title."
+            }), 400
+        
+        if not release_date:
+            return jsonify({
+                "details" : "Request body must include release_date."
+            }), 400
+        
+        if not total_inventory:
+            return jsonify({
+                "details" : "Request body must include total_inventory."
+            }), 400
+        
+        new_video = Video(
+            title=title,
+            release_date=release_date,
+            total_inventory=total_inventory
+        )
+        db.session.add(new_video)
+        db.session.commit()
+
+        return {
+            "id": new_video.id,
+            "title" : new_video.title,
+            "total_inventory" : new_video.total_inventory
+        }, 201
+
+
+@videos_bp.route("/<video_id>", methods= ["GET", "PUT", "DELETE"])
+def handle_video(video_id):
+    if not video_id.isnumeric():
+        return jsonify(None), 400
```

Comment:
The most dependable way to check whether a python string represents an integral value is to try to convert it to an int in a try block, then if it raises a ValueError, we know it wasn't an int, so here, that could look like:

```python
    try:
        int(video_id)
    except ValueError:
        return jsonify(None), 400
```

The string methods like `isnumeric` and `isdigit` are more about detecting classes of characters (think Unicode) than they are for determining what numeric types we might be able to convert the string value into.